### ISSUE TITLE
Steinhafels - ATP Dates

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2444]
_Imported from trac ticket 2444.
Created by **dzuern**
Opened in trac: Aug 12, 2013, 2:55:52 PM
Last modified in trac: Aug 13, 2013, 9:45:48 AM_
***
_by dzuern:_
When No stock is available, display the PO date + 3 days.

For Special Order Product (200000002 or 400000004), the available date is today + 70 days.  This is just to show it into the future until the sku and PO are created.

**Comment by kentbower** [comment]
_**kbower** trac comment on Aug 12, 2013, 4:27:52 PM:_

(See mor.py)
```python
####################################
# PurchaseOrderDetail modifications
def stn_availability_date(purchaseorderdetail, product):
    """
    Purchase orders add 3 days to availablity date
    """
    return purchaseorderdetail.arrivaldate + timedelta(3)
PurchaseOrderDetail.availability_date=stn_availability_date


#########################
# Product modifications
def stn_faith_leaddays(product):
    """
    Return:
    None means assume no atpdate, set to unknown.
    Value means add to today()
    """
    if product.productid in (u'200000002', u'400000004'):
        # Special Order Product
        # This is just to show it into the future until the sku 
        # and PO are created.
        return 70
        # return product.leaddays
    return None
Product.faith_leaddays=stn_faith_leaddays
```

**Comment by kentbower** [comment]
_**kbower** trac comment on Aug 12, 2013, 4:29:16 PM:_

Scott, should be able to add those 2 functions and test.

**Comment by kentbower** [comment]
_**kbower** trac comment on Aug 12, 2013, 4:36:22 PM:_

(It would be better to just `return product.leaddays` for those special order products and ask them to set their data to 70 in the database instead of hard coding 70)

**Comment by kentbower** [comment]
_**sjohnson** trac comment on Aug 13, 2013, 9:45:48 AM:_

(Changeset [[4431]](../commit/f6bf2b8ed0d3fb6ff32b880e82c6bdbc3a418cd5))
* Fixes #2444 Steinhafels ATP date #stn
    * When no stock PO date plus 3
    * For SPO (200000002, 400000004) use product.leaddays

**Comment by kentbower** [comment]
_**sjohnson** trac status change on Aug 13, 2013, 9:45:48 AM:_ **closed**
